nomt/src/beatree/branch/node.rs

```diff
@@ -217,18 +285,43 @@ impl<'a> BranchNodeView<'a> {
         let offset = BRANCH_NODE_SIZE - (self.n() as usize - i) * 4;
         u32::from_le_bytes(self.inner[offset..offset + 4].try_into().unwrap())
     }
+
+    pub fn node_pointers(&self) -> &[[u8; 4]] {
+        let node_pointers_init = BRANCH_NODE_SIZE - self.n() as usize * 4;
+        unsafe {
+            std::slice::from_raw_parts(
+                self.inner[node_pointers_init..].as_ptr() as *const [u8; 4],
+                self.n() as usize,
+            )
+        }
+    }
 }
 
 unsafe impl Send for BranchNode {}
 
 // A RawPrefix is made by a tuple of raw bytes and the relative bit length
 pub type RawPrefix<'a> = (&'a [u8], usize);
-// A RawSeparator is made by a triple, the raw bytes, the bit-offset
-// at which the separator starts to be encoded in the first byte
+
+// Data required to extract one or more separators from the branch node
+#[derive(Debug)]
+struct RawSeparatorsData {
+    // start of the separators within the branch node page
+    start: usize,
+    // length in bytes of the separators, it must be a multiple of 8
+    byte_len: usize,
+    // bit offset of the init of the separators within the first byte
+    bit_init: usize,
+    // bit length of the separators
+    bit_len: usize,
+}
+
+// A RawSeparators is made by a triple, the raw bytes, the bit-offset
+// at which the separators starts to be encoded in the first byte
 // and the relative bit length
 //
 // The raw bytes are always a multiple of 8 bytes in length
-pub type RawSeparator<'a> = (&'a [u8], usize, usize);
+pub type RawSeparators<'a> = (&'a [u8], usize, usize);
```

Comment:
it's odd to define this type as a plural but then use it like a singular in e.g. `fn raw_separator` or `fn reconstruct_key`